nikola/__main__.py

```diff
@@ -238,16 +238,13 @@ def __init__(self, nikola, quiet=False):
         self.quiet = quiet
 
     def load_tasks(self, cmd, opt_values, pos_args):
+        DOIT_CONFIG = config.get('DOIT_CONFIG', {})
         if self.quiet:
-            DOIT_CONFIG = {
-                'verbosity': 0,
-                'reporter': 'zero',
-            }
+            DOIT_CONFIG['verbosity'] = 0
+            DOIT_CONFIG['reporter'] = 'zero'
         else:
-            DOIT_CONFIG = {
-                'reporter': ExecutedOnlyReporter,
-                'outfile': sys.stderr,
-            }
+            DOIT_CONFIG['reporter'] = ExecutedOnlyReporter
```

Comment:
So, it seems like going back to the original and adding this one line:

```
        DOIT_CONFIG.update(config.pop('DOIT_CONFIG', {}))
```
would do what you want.